From 3677ea7f9476458aa6dec440243de3a6fb1343a9 Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Wed, 8 Oct 2014 12:04:00 -0400
Subject: [PATCH] 8250/8250_dw: fix compile failure due to stable/Yocto
 conflict

As of merge 60a9d9fc565e4503dbb8705803e83d906afc4ad2, "Merge
tag 'v3.10.48' into standard/base" the 8250_dw.c fails to
compile due to an undeclared variable.

This happens because stable brought in:

 -------------------------
 commit 6d5e79331417886196cb3a733bdb6645ba85bc42
 Author: Tim Kryger <tim.kryger@linaro.org>
 Date:   Tue Oct 1 10:18:08 2013 -0700

    serial: 8250_dw: Improve unwritable LCR workaround

    commit c49436b657d0a56a6ad90d14a7c3041add7cf64d upstream.

 [...]

    [wangnan: backport to 3.10.43:
      - adjust context
      - remove unneeded local var]
    Signed-off-by: Wang Nan <wangnan0@huawei.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
 ------------------------

...which deletes the p->private_data declaration since it became
unused at that point, however in Yocto, we also have this:

 -----------------------
 commit 0e02b050c3cafbcbf9952125089a27e02d6ecea9
 Author: David Daney <david.daney@cavium.com>
 Date:   Wed Jun 19 20:37:27 2013 +0000

    tty/8250_dw: Add support for OCTEON UARTS.

 [...]

    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
    (cherry picked from commit d5f1af7ece96cf52e0b110c72210ac15c2f65438)
    Signed-off-by: Darren Hart <dvhart@linux.intel.com>
 -----------------------

...which _adds_ another user of the p->private_data.

Here we restore the declaration in order that 8250_dw compiles.

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
[PG: add root cause info to commit log.]
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 drivers/tty/serial/8250/8250_dw.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/tty/serial/8250/8250_dw.c b/drivers/tty/serial/8250/8250_dw.c
index 36fe9d9262c0..5caf10ee14b1 100644
--- a/drivers/tty/serial/8250/8250_dw.c
+++ b/drivers/tty/serial/8250/8250_dw.c
@@ -217,6 +217,7 @@ static unsigned int dw8250_serial_in32(struct uart_port *p, int offset)
 
 static int dw8250_handle_irq(struct uart_port *p)
 {
+	struct dw8250_data *d = p->private_data;
 	unsigned int iir = p->serial_in(p, UART_IIR);
 
 	if (serial8250_handle_irq(p, iir)) {
-- 
1.8.1.2

